### PR TITLE
ux specs for product overview page

### PR DESCRIPTION
**Pull request by deernano**
**What this PR does / why we need it**:

> Add UX specifications for the Product overview page
JIRA: https://issues.redhat.com/browse/THREESCALE-5420

### CODE REVIEW

File: portafly/doc/ux_specs/product/overview.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+* [Mockups](https://marvelapp.com/prototype/ee82j74/screen/70939434)
+* [Mockups JIRA ticket](https://issues.redhat.com/browse/THREESCALE-4389)
+* [Parent JIRA ticket](https://issues.redhat.com/browse/THREESCALE-4197)
+
+##### Page header
+* Edit button
+  * Secondary button
+* Delete button
+  * Customized tertiary button
```
Comment (josemigallas) [inline portafly/doc/ux_specs/product/overview.md:13]: What's the customization about? Only color? If so, could you add its code/name?

Comment (deernano) [reply portafly/doc/ux_specs/product/overview.md:13]: @cntlsn Should we use the warning color for this? and the secondary button style?

Comment (cntlsn) [reply portafly/doc/ux_specs/product/overview.md:13]: @josemigallas the customization is eventually a temporary thing. a **destructive secondary button** is in the process of being implemented by the patternfly team (https://github.com/patternfly/patternfly/issues/1094#issuecomment-470592366). in the meantime we could just use a primary destructive button, or if easy enough customize a secondary button to resemble a destructive button, aka creating a custom class such as:
```
.pf-c-button.pf-m-secondary-danger {
    --pf-c-button--after--BorderColor: var(--pf-global--danger-color--100);
    color: var(--pf-global--danger-color--100);
    background-color: var(--pf-c-button--m-secondary--BackgroundColor);
}
```

File: portafly/doc/ux_specs/product/overview.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+##### Delete modal ([Global](../../global_components/modal.md))
+* [PatternFly Modal dialog design guidelines](https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/modal)
```
Comment (cntlsn) [inline portafly/doc/ux_specs/product/overview.md:16]: @deernano this should cover the specs for the delete modal (feel free to edit):
```
#### Delete modal ([specs](../../global_components/modal.md))
* Modal's title will include the name of the current Product the user is trying to delete
* Enter product name > text input field [required]
  * Validation error:
    * Wrong product name on _onblur_
* Buttons
  * Delete button [Disabled state until all conditions required to submit the form are met]
    * Primary danger button variant
    * Submits form
  * Cancel button [Always enabled]
    * Dismisses modal
```

@josemigallas please review.

**Note:** we should probably move some of these specs to `modal.md` so they can be reused for other contexts (backend deletion, user deletion etc), and only add the specs specific to this modal in this file

Comment (deernano) [reply portafly/doc/ux_specs/product/overview.md:16]: Updated to Marvel: https://marvelapp.com/prototype/ee82j74/screen/71008634

File: portafly/doc/ux_specs/index.md [hunk 3]
```diff
@@ -22,6 +22,7 @@ Please refer to the single page files together with mockups linked in each page
 ##### Product
 * Applications
   * [Applications index](./product/applications/index.md)
+* [Overview](./product/index.md)
```
Comment (josemigallas) [inline portafly/doc/ux_specs/index.md:25]: Do we want the indexes in the same order as in the sidebar navigation? If so I guess Overview goes first.
```suggestion
* [Overview](./product/index.md)
* Applications
  * [Applications index](./product/applications/index.md)
```

Comment (deernano) [reply portafly/doc/ux_specs/index.md]: Updated

**Pull request merged**